src/pages/_app.tsx

```diff
@@ -64,6 +67,23 @@ function MyApp({ Component, pageProps }): JSX.Element {
     };
   }, [router.events]);
 
+  // Redirect to complete signup page if the user profile is not complete
```

Comment:
Is there a better way to do the same with middlewares than this way?

Reply:
Middleware's are next.js feature and not dependent on Vercel. https://nextjs.org/docs/app/building-your-application/routing/middleware#platform-support